src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemCsv.scala

```diff
@@ -0,0 +1,51 @@
+package nl.knaw.dans.easy.stage.fileitem
+
+import java.io.InputStream
+
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVParser.parse
+import org.rogach.scallop.TrailingArgsOption
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+import scala.io.Source.fromInputStream
+import scala.util.{Success, Try}
+
+object FileItemCsv {
+  val log = LoggerFactory.getLogger(getClass)
+
+  def read(in: InputStream, conf: FileItemConf): Try[Array[FileItemSettings]] = Try {
+    val trailArgs: List[String] = List(conf.sdoSetDir.apply().toString)
+    val csv = {
+      val rawContent = fromInputStream(in).mkString
+      parse(rawContent, CSVFormat.RFC4180).getRecords.map(_.toList)
+    }
+    if (csv.isEmpty){
+      log.warn("CSV file empty")
+      return Success(Array[FileItemSettings]())
+    }
+    val head = csv.head
+    validateHeaders(conf,head)
+    csv.tail.map(value =>
+      // a key-value pair per column for an input line
+      head.map(key => key.toLowerCase).zip(value)
+      // ignore columns with headers that are not options
+      .filter(t => !head.contains(t._1))
+      // convert to command line options
+      .flatMap(t => Array("--" + t._1, t._2))
+    ).filter(args => args != Nil).map(args =>
+      FileItemSettings(Array(args, trailArgs).flatten)
```

Comment:
Concatenate lists like this:   

    args ++ trailArgs

First creating an array of lists and then flattening it is, while yielding the same result, a bit harder to read.